### PR TITLE
OCM-2373 | feat: Moved associated to GetRole function to common

### PR DESCRIPTION
**Pull request by den-rgb**
Related to: https://github.com/openshift/rosa/pull/1547

## JIRA 
[OCM-2373](https://issues.redhat.com/browse/OCM-2373)

### CODE REVIEW

**Comment by oriAdler** [comment]
Checking the ROSA repo, and some files in this repo, it seems that our convention for `go` file name is snakeCase.
Reference https://medium.com/@kdnotes/golang-naming-rules-and-conventions-8efeecd23b68

File: pkg/ocm/validations/getRole.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package validations
+
+import (
+	"net/url"
+	"strings"
+
+	errors "github.com/zgalor/weberr"
+)
+
+func ValidateIssuerUrlMatchesAssumePolicyDocument(
```
Comment (oriAdler) [inline pkg/ocm/validations/getRole.go:10]: This function doesn't match the file name, maybe we should for a more generic naming for the file?

File: pkg/aws/validations/getRole.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    for _, tag := range roleTags {
+        if aws.StringValue(tag.Key) == tags.ManagedPolicies && aws.StringValue(tag.Value) == "true" {
+            return true
+        }
+    }
+
+    return false
+}
+
+func HasCompatibleVersionTags(iamTags []*iam.Tag, version string) (bool, error) {
```
Comment (oriAdler) [inline pkg/aws/validations/getRole.go:29]: Also here, maybe we need more generic file names.
As this function is not related to a `getRole` AWS call.

Comment (den-rgb) [reply pkg/aws/validations/getRole.go:29]: @oriAdler what naming do you suggest?

File: go.mod [hunk 3]
```diff
@@ -9,11 +9,19 @@ require (
...
+
+require (
+	github.com/aws/aws-sdk-go v1.45.25
 	github.com/go-logr/logr v1.2.4 // indirect
 	github.com/go-task/slim-sprig v0.0.0-20230315185526-52ccab3ef572 // indirect
 	github.com/google/go-cmp v0.5.9 // indirect
 	github.com/google/pprof v0.0.0-20210407192527-94a9f03dee38 // indirect
+	github.com/openshift/rosa v1.2.27
 	github.com/stretchr/testify v1.7.0 // indirect
+	github.com/zgalor/weberr v0.8.2
```
Comment (ciaranRoche) [inline go.mod:24]: I dont think we need weberr in common, as its not used in ROSA. 

We can just return a generic error, and leave CS wrap in weberr as needed.

File: pkg/aws/validations/getRole.go (hunk 2 above)
Comment (oriAdler) [reply pkg/aws/validations/getRole.go:29]: Maybe `iam` can work here? as all validations are related to either role or role tags.
